src/evaluation.ts

```diff
@@ -38,6 +38,11 @@ export const evaluate = ({
     const {variableId, operator, value: comparedValue} = expression;
 
     validateRuleInvoke({operator, comparedValue});
+    const passedValue = variableIdToValuesMap.get(variableId);
+
+    if (passedValue === undefined) {
+      return [false];
+    }
 
     return [
       ruleHandlers[operator]({
```

Comment:
you don't need to get again from the map `passedValue: variableIdToValuesMap.get(variableId),` as 1st parameter of ruleHandlers[operator], since you already got it

Reply:
yep, my bad